packages/experimental/RadioGroup/src/Radio/useRadio.ts

```diff
@@ -25,25 +30,22 @@ export const useRadio = (props: RadioProps): RadioState => {
     ...rest
   } = props;
 
-  // Grabs the context information from RadioGroup (currently selected button and client's onChange callback)
-  const selectedInfo = useRadioGroupContext();
-
   const buttonRef = useViewCommandFocus(componentRef);
 
   /* We don't want to call the user's onChange multiple times on the same selection. */
   const changeSelection = React.useCallback(() => {
-    if (value != selectedInfo.value) {
-      selectedInfo.onChange && selectedInfo.onChange(value);
-      selectedInfo.updateSelectedButtonRef && componentRef && selectedInfo.updateSelectedButtonRef(componentRef);
+    if (value !== radioGroupContext.value) {
+      radioGroupContext.onChange && radioGroupContext.onChange(value);
+      radioGroupContext.updateSelectedButtonRef && componentRef && radioGroupContext.updateSelectedButtonRef(componentRef);
     }
-  }, [selectedInfo, value, componentRef]);
+  }, [radioGroupContext, value, componentRef]);
 
   /* We use the componentRef of the currently selected button to maintain the default tabbable
     element in a RadioGroup. Since the componentRef isn't generated until after initial render,
     we must update it once here. */
   React.useEffect(() => {
-    if (value == selectedInfo.value) {
-      selectedInfo.updateSelectedButtonRef && componentRef && selectedInfo.updateSelectedButtonRef(componentRef);
+    if (value === radioGroupContext.value) {
+      radioGroupContext.updateSelectedButtonRef && componentRef && radioGroupContext.updateSelectedButtonRef(componentRef);
     }
   }, []);
```

Comment:
Post review comment, but I see an eslint error about the empty dependency array on line 48. Should it be empty?

Reply:
Yes, it should be empty here since we only want to update only once. I believe this is an error that occurs for old RadioGroup too.